IOS only upgrade react-native to 0.72.x

siddarthkay opened the pull request:
This PR is just made to test and fix effects of this upgrade on CI.
Note: This PR only contains IOS side of the upgrades.
state: Ready

siddarthkay committed:
upgrade react-native to 0.70.12

siddarthkay commented on nix/deps/nodejs-patched/default.nix:
```diff
@@ -49,12 +47,7 @@ stdenv.mkDerivation {
   '';
-  # Do not add a BuildId to the generated libraries, for reproducibility
-  patchBuildIdPhase = ''
-    substituteInPlace ./node_modules/react-native/ReactAndroid/src/main/jni/Application.mk --replace \
```
now that `ReactAndroid/src/main/jni/Application.mk` no longer exists, We'll have to figure out where to update build-id :)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | d68c598e | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-16721/1/) | 2023-07-19 15:09:13 | ~3 min | `android-e2e` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-16721/1/consoleText) |

siddarthkay committed:
chore: drop Android specific Jenkinsfiles

siddarthkay committed:
upgrade react-native to 0.71.12

siddarthkay committed:
upgrade react-native-reanimated to 3.0.0

siddarthkay committed:
upgrade react-native-redash to 18.1.0

- We need to do this to get rid of react-native-reanimated v1 usage since its incompatible with v3.
- Temporarily modifying imports
at `src/quo/animated.cljs` and at `src/mocks/js_dependencies.cljs`
- A thorough cleanup is required in a follow up.

siddarthkay committed:
upgrade react-native-draggable-flatlist to 4.0.1

- We need to do this to because versions of library `react-native-draggable-flatlist` older than `4.0.0` rely on `react-native-reanimated ` v1 and that does not work for us if we wish to upgrade `react-native` further.

siddarthkay commented:
Hit a blocker at `react-native` version `0.71.12`

Although builds are green at the moment but the app indefinitely stays on the splash screen and when you put the app in background and open it again you are greeted with this not so specific error.

![Simulator Screenshot - iPhone 11 Pro - 2023-07-20 at 09 32 58](https://github.com/status-im/status-mobile/assets/64726664/5b9778ce-3aba-4691-8704-9914553b39b8)

My current hypothesis is that I need to more thoroughly clean up all of the v1 usage of `react-native-redash` library which in turn uses the v1 of `react-native-reanimated` library.

More updates on this later.

siddarthkay committed:
upgrade `react-native` to `0.72.3`

siddarthkay committed:
(0.72.3) : remove `patchHermesPhase`

`./node_modules/react-native/react.gradle` path no longer exists in `react-native` 0.72.3

siddarthkay committed:
(0.72.3) : bump `react-native-reanimated` to 3.4.1

siddarthkay committed:
(0.72.3) : semi aggressively nuke old code

siddarthkay commented:
I briefly had this error after the commit for re-animated upgrade.

![Screenshot_2023-08-03_at_7 02 46_PM](https://github.com/status-im/status-mobile/assets/64726664/54598846-2e66-4701-9ba8-0f678c61f3d4)

siddarthkay commented:
However I am back with my old nemesis and now this is now the latest error.
It seems to be reagent related. 
I hate how this does not give out any valuable information.

<img width="1053" alt="Screenshot 2023-08-13 at 6 34 17 PM" src="https://github.com/status-im/status-mobile/assets/64726664/eb765bf7-12c3-4665-ac23-5bd204dcdedb">

siddarthkay committed:
(0.72.3) fix for getting past splash screen

siddarthkay commented:
IOS Side is ready, time to focus on Android side

<img width="923" alt="Screenshot 2023-08-19 at 9 39 56 AM" src="https://github.com/status-im/status-mobile/assets/64726664/2b846238-fc06-4d37-8850-7bc54fba4ce8">

siddarthkay committed:
fixed `node-libs-react-native/globals` import

siddarthkay committed:
upgrade react-native to 0.72.4

siddarthkay committed:
fix: re-animated v3 animations cleanup

- Easing Import
- FadeIn swap for Transition Import
- easing
- linear
- Extrapolate
- Interpolate

siddarthkay committed:
fix: node deps mess during rebase :)
